scene/gui/tab_bar.cpp

```diff
@@ -1520,6 +1550,9 @@ void TabBar::ensure_tab_visible(int p_idx) {
 	if (!is_inside_tree() || !buttons_visible) {
 		return;
 	}
+	if (p_idx == -1) {
```

Comment:
Shouldn't this check for `deselect_enabled`?
Also might be worth mentioning it in the docs, as this is rather unexpected.

Reply:
I don't want it to error when -1 is passed whether `deselect_enabled` is true or false, since it is a value that current_tab can be.
I'm not sure what is unexpected about it, it wouldn't do anything otherwise but error.

Reply:
> since it is a value that current_tab can be.

But it can be -1 only when `deselect_enabled` or there are no tabs, right? When -1 `current_tab` results in an error, so should this method I think.

Reply:
Ok, I'll have to have another check for no valid tabs then.

Reply:
Maybe it can be a method, since you need it a few times. Something like `can_deselect()`.